src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+// class Storage - handles the loading of data when duke is run and
+// saving of data when duke is closed
+class Storage {
+    private String filePath;
+
+    public Storage(String s) {
+        this.filePath = s;
+    }
+
+    public ArrayList<Task> load() {
+        System.out.println("    . . . Loading . . . ");
+        File file = new File(this.filePath);
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        File dir = new File(System.getProperty("user.dir") + "/data");
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+        try {
+            Scanner saveFile = new Scanner(file);
+            System.out.println("    Saved data found, welcome back!");
+            while (saveFile.hasNextLine()) {
+                loadSaved(saveFile.nextLine(), tasks);
+            }
+        } catch (FileNotFoundException ex) {
+            System.out.println("    No saved data not found, new file will be created");
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+        return tasks;
+    }
+
+    public static void loadSaved(String s, ArrayList<Task> tasks) {
+        if (s.substring(0, 1).equals("T")) {
+            Todo newTodo = new Todo(s.substring(8));
+            if (s.substring(4, 5).equals("1")) {
+                newTodo.toggleMarked();
+            }
+            tasks.add(newTodo);
+
+        } else if (s.substring(0, 1).equals("D")) {
+            String marked = s.substring(4, 5);
+            String task = s.substring(8);
+            Deadline newDeadline = new Deadline(task.substring(0, task.indexOf("|") - 1),
+                    task.substring(task.indexOf("|") + 2));
+            if (marked.equals("1")) {
+                newDeadline.toggleMarked();
+            }
+            tasks.add(newDeadline);
+
+        } else {
+            String marked = s.substring(4, 5);
+            String task = s.substring(8);
+            String desc = task.substring(0, task.indexOf("|") - 1);
+            task = task.substring(task.indexOf("|") + 2);
+            Event newEvent = new Event(desc, task.substring(0, task.indexOf("|") - 1),
+                    task.substring(task.indexOf("|") + 2));
+            tasks.add(newEvent);
+        }
+    }
```

Comment:
From what I understand, your code will modify the `ArrayList<Task> tasks` argument in place with a void return type. What is the reason for doing it this way instead of having the function return an `ArrayList<Task>` or a new `TaskList`?